### PR TITLE
feat(carousel): added carousel options

### PR DESCRIPTION
**Pull request by shinrox**
This PR enable some options on carousel component

## Focus
Whether to focus or not on carousels without an interval (auto play)

## Wrap
Defines if carousel cycles continuously or have hard stops

## Pause
Whether the carousel should stop on hover or not

### CODE REVIEW

**Comment by tmorehouse** [comment]
Could you change the new props to `noFocus`, `noWrap` and `noHoverPause`, and default them to `false`?

This would better follow other components (when disabling the defaults), and works better when using just the presence of the prop to disable the default.

**Comment by shinrox** [comment]
Instead of changing what about adding those you've mentioned?

Except for the focus (cause it doesn't exists in bootstrap) the other ones are following the bootstrap default, should I really change them? Not following the framework that this plugin is built on is a bit confusing, isn't it?

https://getbootstrap.com/docs/4.1/components/carousel/#options

**Comment by tmorehouse** [comment]
@shinrox Its more of a Vue thing.

Look at it this way when adding attributes... such as `wrap` which defaults to `true`:

```html
/* : required to convert string to boolean */
<b-carousel :wrap="false" ... >

 /* actually evaluates to a `truthy` value, so in fact wrap is true, which is default */
<b-carousel wrap="false" ... >

 /* ends up being the same as default (Boolean true) */
<b-carousel wrap ... >
```

Where as the `no-wrap` option is more intuitive to disable the default behaviour:
```html
/* not attribute value required to disable default */
<b-carousel no-wrap  ... >

 /* actually evaluates to a `truthy` value */
<b-carousel no-wrap="true" ... >

 /* evaluates to a `true` value because of v-bind: */
<b-carousel :no-wrap="true" ... >
```

It makes the attribute value processing much simpler (and the components smaller)

Internally, with no attribute/props
- focus functionality default is true (required for ARIA compliance)
- wrap functionality default is true
- pause functionality default is true (required for ARIA compliance)

Hence it makes more sense to use the `no-` form for the props (as is the norm with all other Bootstrap Vue components where `true` is the default)

**Comment by shinrox** [comment]
Cool, thanks for the explanation, I'll try to update this PR by the end of this week.

File: src/components/carousel/README.md [hunk 1]
```diff
@@ -11,6 +11,9 @@ controls and indicators. Please be aware that nested carousels are **not** suppo
                 style="text-shadow: 1px 1px 2px #333;"
                 controls
                 indicators
+                focus
+                wrap
```
Comment (tmorehouse) [inline src/components/carousel/README.md]: using just `wrap` is the same thing as not using `wrap` at all... this makes for confusing props and docs.

File: src/components/carousel/README.md [hunk 2]
```diff
@@ -117,6 +118,16 @@ Set the prop `indicators` to show the slide indicator buttons.
 
 Both indicators and controls can be set at the same time or independently.
 
+# Focus
+The carousel will focus the next slide for screen readers if not in play mode (doesn't have an interval)
+
+To avoid this, set the prop `focus` to false
```
Comment (tmorehouse) [inline src/components/carousel/README.md]: this is not recommended as it breaks ARIA and accessibility. 

It actually focuses the _current_ slide not the next one that isn't shown.

File: src/components/carousel/README.md [hunk 3]
```diff
@@ -11,6 +11,9 @@ controls and indicators. Please be aware that nested carousels are **not** suppo
                 style="text-shadow: 1px 1px 2px #333;"
                 controls
                 indicators
+                focus
+                wrap
+                pause="hover"
```
Comment (tmorehouse) [inline src/components/carousel/README.md]: why is this a string, when the only option is really  on or off (true or false)? And it is either on hover or not.

File: src/components/carousel/carousel.js [hunk 4]
```diff
@@ -413,11 +433,11 @@ export default {
         setAttr(nextSlide, 'aria-hidden', 'false')
         currentSlide.tabIndex = -1
         nextSlide.tabIndex = -1
-        if (!this.isCycling) {
+        if (!this.isCycling && this.focus) {
           // Focus the next slide for screen readers if not in play mode
           nextSlide.tabIndex = 0
           this.$nextTick(() => {
-            nextSlide.focus()
```
Comment (tmorehouse) [inline src/components/carousel/carousel.js]: normally the focusing of the slide only happens when the slide is paused due to a mouseover.

**Comment by tmorehouse** [comment]
Any movement on this?

**Comment by tmorehouse** [comment]
Closing as stale

**Comment by qkdreyer** [comment]
@shinrox Are you willing to finish this PR ?

**Pull request closed without merging**